### PR TITLE
refactor: replace `ColdHeader` with `CodeSignature`

### PR DESCRIPTION
**Pull request by Tropix126**
## Describe the changes this PR makes. Why should it be merged?
- Replaces the previous `ColdHeader` type with a higer-level `CodeSignature` type.
- Update to `vex-sdk` 0.17.0.

## Additional Context
Requires testing to ensure boot flags work as intended.

### CODE REVIEW

File: packages/vexide-startup/src/lib.rs [hunk 1]
```diff
@@ -13,41 +13,78 @@
...
     }
 }
 
+/// Program Code Signature
+///
+/// The first 16 bytes of a VEX user code binary contain a user code signature,
+/// containing some basic metadata and startup flags about the program. This
+/// signature must be at the start of the binary for booting to occur.
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq)]
+pub struct CodeSignature(vex_sdk::vcodesig, [u32; 4]);
```
Comment (doinkythederp) [inline packages/vexide-startup/src/lib.rs:65]: What's the purpose of the extra 4 bytes? Are they just padding to get to 0x20?

File: packages/vexide-startup/src/lib.rs [hunk 2]
```diff
@@ -13,41 +13,78 @@
...
+
+bitflags! {
+    /// Program Flags
+    ///
+    /// These bitflags are part of the [`CodeSignature`] that determine some small
+    /// aspects of program behavior when running under VEXos. This struct contains
+    /// the flags with publicly documented behavior.
+    #[derive(Default, Debug, Clone, Copy, Eq, PartialEq)]
+    pub struct ProgramFlags: u32 {
+        /// Default graphics colors (foreground/background) will be inverted.
```
Comment (doinkythederp) [inline packages/vexide-startup/src/lib.rs:47]: From my testing these display inversion flags don't affect foreground color (it always starts as `#c0c0ff`).

File: packages/vexide-startup/src/lib.rs [hunk 3]
```diff
@@ -13,41 +13,78 @@
...
+        const INVERT_DEFAULT_GRAPHICS = 1 << 0;
+
+        /// VEXos scheduler simple tasks will be killed when the program requests exit.
+        const KILL_TASKS_ON_EXIT = 1 << 1;
+
+        /// Default graphics colors (foreground/background) will invert based on
+        /// the selected system theme.
+        const THEMED_DEFAULT_GRAPHICS = 1 << 2;
```
Comment (doinkythederp) [inline packages/vexide-startup/src/lib.rs:55]: It might be good to specify than when both inversion options are specified, the themed one "wins".

**Review by doinkythederp — DISMISSED** [review]

**Comment by Tropix126** [comment]
Latest changes need testing

File: packages/vexide-startup/link/v5.ld [hunk 4]
```diff
@@ -1,52 +1,56 @@
...
+    USER_RAM : ORIGIN = __user_ram_start, LENGTH = __user_ram_length
 }
 
 __stack_length = 0x400000;
-/*
-I'm not sure why subtracting anything is necessary, but it fixes memory permission errors.
-0x100 is an arbitrary number that works.
-*/
-__heap_end = __cold_end - __stack_length - 0x100;
+__heap_end = __user_ram_end - __stack_length - 0x100;
```
Comment (Gavin-Niederman) [inline packages/vexide-startup/link/v5.ld:14]: I think its worth keeping the documentation for this subtraction.

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
Just this.

**Review by Gavin-Niederman — APPROVED** [review]
LGTM

**Pull request merged**